### ISSUE TITLE
Question about the order of simulation and real data analyses in SDePER study

### ISSUE THREAD

**Issue by ThesecondtoNone** [az7jh2/SDePER#17]
Sorry to bother you again. May I ask whether you first ran the simulation datasets or the real datasets at that time? Did you tune the parameters on the simulation datasets first and then apply them to the real datasets?

**Comment by az7jh2** [comment]
Hi.

We ran SDePER independently on simulated data and real datasets.

Since SDePER includes many parameters, or so-called hyperparameters, I am not sure which specific parameters you are referring to. For the most important parameters related to CVAE, we summarized some recommendations based on our knowledge and experience from testing SDePER. You can find them [here](https://sdeper.readthedocs.io/en/latest/best_practice.html#set-cvae-related-options).

For the hyperparameters related to the adaptive lasso and graph Laplacian penalty, the only two hyperparameters in GLRM component, SDePER uses embedded cross-validation to identify the optimal values. When `--diagnosis` is set to `true`, the cross-validation loss curve will be plotted. You can check whether the curve shows a clear U-shape. Otherwise, you may need to adjust the search ranges of these hyperparameters to define a more reasonable range (please refer [GLRM hyperparameter-related options](https://sdeper.readthedocs.io/en/latest/deconvolution_opt.html#glrm-hyperparameter-related-options) for more details).

For some CVAE-related hyperparameters, since CVAE is essentially a neural network, we allow users to specify certain settings, such as the number of hidden layers. At the same time, SDePER adaptively determines some parameters, such as the number of neurons in each hidden layer, while fixing others. For example, the number of neurons in the latent layer is fixed as three times the number of cell types, and the activation functions are also fixed.
